bundles/core/src/main/java/com/adobe/cq/commerce/core/components/internal/models/v1/navigation/GraphQLCategoryProvider.java

```diff
@@ -18,54 +18,54 @@
 import java.util.List;
 import java.util.function.Function;
 
-import com.adobe.cq.commerce.graphql.client.GraphqlClient;
-import com.adobe.cq.commerce.graphql.client.GraphqlRequest;
-import com.adobe.cq.commerce.graphql.client.GraphqlResponse;
-import com.adobe.cq.commerce.magento.graphql.Query;
-import com.adobe.cq.commerce.magento.graphql.gson.Error;
-import com.adobe.cq.commerce.magento.graphql.gson.QueryDeserializer;
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
 
-
-import com.adobe.cq.commerce.core.components.internal.models.v1.Utils;
+import com.adobe.cq.commerce.core.components.internal.models.v1.MagentoGraphqlClient;
+import com.adobe.cq.commerce.graphql.client.GraphqlRequest;
+import com.adobe.cq.commerce.graphql.client.GraphqlResponse;
 import com.adobe.cq.commerce.magento.graphql.CategoryTree;
 import com.adobe.cq.commerce.magento.graphql.CategoryTreeQuery;
 import com.adobe.cq.commerce.magento.graphql.CategoryTreeQueryDefinition;
 import com.adobe.cq.commerce.magento.graphql.Operations;
+import com.adobe.cq.commerce.magento.graphql.Query;
 import com.adobe.cq.commerce.magento.graphql.QueryQuery;
+import com.adobe.cq.commerce.magento.graphql.gson.Error;
 import com.day.cq.wcm.api.Page;
 
 class GraphQLCategoryProvider {
+
     private static final Logger LOGGER = LoggerFactory.getLogger(GraphQLCategoryProvider.class);
+
     private static final CategoryTreeQueryDefinition categoryQueryDefinition = query -> {
         Function<CategoryTreeQuery, CategoryTreeQuery> categoriesQuery = q -> q.id().name().urlPath().position();
         categoriesQuery.apply(query).children(categoriesQuery::apply);
     };
-    private GraphqlClient graphqlClient;
+
+    private MagentoGraphqlClient client;
 
     GraphQLCategoryProvider(Page page) {
-        this.graphqlClient = page.getContentResource().adaptTo(GraphqlClient.class);
-        if (graphqlClient == null) {
-            LOGGER.warn("GraphQL client not available for page: " + page.getPath());
+        client = new MagentoGraphqlClient(page.getContentResource());
+        if (client == null) {
```

Comment:
This condition is always false. Even in the situation when your MagentoGraphqlClient instance is invalid (its internal GraphqlClient is null) and the old code would detect client == null and act accordingly. 
So, the error handling logic is invalid and you will get random NPE-s later.